ci/integTestGen/src/integTestGen.jl

```diff
@@ -0,0 +1,265 @@
+- `modified_url`: Stores the Git url set by the environment variable.
+- `env_var`: Name of the environment variable to set the modified_url.
+"""
+mutable struct PackageInfo
+    url::String
+    modified_url::String
+    env_var::String
+    PackageInfo(url, env_var) = new(url, "", env_var)
+end
+
+package_infos = Dict(
+    "QED" => PackageInfo(
+        "https://github.com/QEDjl-project/QED.jl.git",
+        "CI_INTG_PKG_URL_QED"),
+    "QEDfields" => PackageInfo(
+            "https://github.com/QEDjl-project/QEDfields.jl.git",
+            "CI_INTG_PKG_URL_QEDfields"),
+    "QEDbase" => PackageInfo(
+            "https://github.com/QEDjl-project/QEDbase.jl.git",
+            "CI_INTG_PKG_URL_QEDbase"),
+    "QEDevents" => PackageInfo(
+            "https://github.com/QEDjl-project/QEDevents.jl.git",
+            "CI_INTG_PKG_URL_QEDevents"),
+    "QEDprocesses" => PackageInfo(
+            "https://github.com/QEDjl-project/QEDprocesses.jl.git",
+            "CI_INTG_PKG_URL_QEDprocesses"),
+)
+
+"""
+    extract_env_vars_from_git_message!()
+
+    Parse the commit message, if set via variable (usual `CI_COMMIT_MESSAGE`) and set custom URLs.
+"""
+function extract_env_vars_from_git_message!(var_name = "CI_COMMIT_MESSAGE")
+    if haskey(ENV, var_name)
+        for line in split(ENV[var_name], "\n")
+            line = strip(line)
+            for pkg_info in values(package_infos)
+                if startswith(line, pkg_info.env_var * ": ")
+                    ENV[pkg_info.env_var] = SubString(line, length(pkg_info.env_var * ": ") + 1)
+                end
+            end
+        end
+    end
+end
+
+"""
+    set_modified_package_url!()
+
+Iterate over all entries of package_info. If an environment variable exits with the same name like,
+the `env_var` entry, set the value of the environment variable to `custom_url`.
+"""
+function set_modified_package_url!()
+    for package_info in values(package_infos)
+        if haskey(ENV, package_info.env_var)
+            package_info.modified_url = ENV[package_info.env_var]
+        end
+    end
+end
+
+"""
+    get_modified_package()::String
+
+Read the name of the modified (project) package from the environment variable `CI_DEPENDENCY_NAME`.
+
+# Returns
+- The name of the modified (project) package
+"""
+function get_modified_package()::String
+    for env_var in ["CI_DEPENDENCY_NAME", "CI_PROJECT_DIR"]
+        if !haskey(ENV, env_var)
+            error("Environment variable $env_var is not set.")
+        end
+    end
+
+    if !haskey(package_infos, ENV["CI_DEPENDENCY_NAME"])
+        package_name = ENV["CI_DEPENDENCY_NAME"]
+        error("Error unknown package name $package_name}")
+    else
+        return ENV["CI_DEPENDENCY_NAME"]
+    end
+end
+
+"""
+    get_depending_projects(package_name, package_prefix, project_tree)
+
+Returns of a list of packages, which has the package `package_name` as dependency. Ignore all packages, which does not start with `package_prefix`.
```

Comment:
Return a list of packages, which has the package `package_name` as a dependency.